fiftyone/utils/torch.py

```
@@ -1847,20 +1843,101 @@ def _is_string_array(targets):
+
+def _shared_list_prefix_len(dims_a, dims_b):
+    """Length of the longest common prefix of two :func:`_list_dims` results."""
+    n = 0
+    for a, b in zip(dims_a, dims_b):
+        if a != b:
+            break
+        n += 1
+    return n
+
+
+def _flatten_with_coords(nested, prefix=()):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `_flatten_with_coords`

(ANN202)